Source/Core/VideoCommon/VertexShaderGen.cpp

```diff
@@ -362,6 +362,16 @@ ShaderCode GenerateVertexShaderCode(APIType api_type, const vertex_shader_uid_da
       break;
     }
 
+    // When q is 0, the GameCube appears to have a special case
+    // This can be seen in devkitPro's neheGX Lesson08 example for Wii
+    // Makes differences in Rogue Squadron 3 (Hoth sky) and The Last Story (shadow culling)
+    // TODO: check if this only affects XF_TEXGEN_REGULAR
+    if (texinfo.texgentype == XF_TEXGEN_REGULAR)
+    {
+      out.Write("if(o.tex%d.z == 0.0)\n", i);
+      out.Write("\to.tex%d.xy = clamp(o.tex%d.xy, float2(-2.0,-2.0), float2(2.0,2.0));\n", i, i);
```

Comment:
What do you think about .z = 1 here, and to drop this special case in the pixel shader?